pkg/virt-handler/cmd-client/client.go

```diff
@@ -260,8 +260,16 @@ func FindPodDirOnHost(vmi *v1.VirtualMachineInstance) (string, error) {
 	return "", fmt.Errorf("No command socketdir for vmi %s", vmi.UID)
 }
 
-// gets the cmd socket for a VMI
-func FindSocketOnHost(vmi *v1.VirtualMachineInstance) (string, error) {
+// SocketFinder represents an interface for finding sockets on the host.
+type SocketFinder interface {
+	FindSocketOnHost(vmi *v1.VirtualMachineInstance) (string, error)
+}
+
+// DefaultSocketFinder is an implementation of SocketFinder that performs the actual socket discovery
+type NewDefaultSocketFinder struct{}
```

Comment:
Sorry for not being clear earlier. When I was asking for a constructor I mean something like this:
```suggestion
type DefaultSocketFinder struct{}

func NewDefaultSocketFinder() SocketFinder{
	return &DefaultSocketFinder{}
}
```

so that instead of using:
```
finder := &cmdclient.DefaultSocketFinder{}
socket, err := finder.FindSocketOnHost(vm)
```
the DefaultSocketFinder can be used as
```
finder := cmdclient.NewDefaultSocketFinder()
socket, err := finder.FindSocketOnHost(vm)
```
**OR** with one single line `socket, err := cmdclient.NewDefaultSocketFinder().FindSocketOnHost(vm)`
What do you think? Thank you!